src/rust/kanto-tui/src/kanto_api.rs

```diff
@@ -0,0 +1,140 @@
+
+pub async fn get_connection(socket_path: &str) -> Result<ClientChannel> {
+    let _path = socket_path.to_owned();
+    let channel = Endpoint::try_from("http://[::]:50051")?
+        .connect_with_connector(service_fn(move |_: Uri| UnixStream::connect(_path.clone())))
+        .await?;
+    Ok(cm_rpc::containers_client::ContainersClient::new(channel))
+}
+
+pub async fn list_containers(channel: &mut ClientChannel) -> Result<Vec<cm_types::Container>> {
+    let _r = tonic::Request::new(cm_rpc::ListContainersRequest {});
+    let containers = channel.list(_r).await?.into_inner();
+    Ok(containers.containers)
+}
+
+pub async fn create_container(
+    channel: &mut ClientChannel,
+    name: &str,
+    registry: &str,
+) -> Result<CreateContainerResponse> {
+    let mut template: Container = serde_json::from_str(CONT_TEMPLATE)?;
+    template.name = String::from(name);
+    template.image.as_mut().ok_or("Field name missing")?.name = String::from(registry);
+
+    let _r = tonic::Request::new(cm_rpc::CreateContainerRequest {
+        container: Some(template),
+    });
+    let _response = channel.create(_r).await?;
+    Ok(_response.into_inner())
+}
+
+pub async fn get_container_by_name(channel: &mut ClientChannel, name: &str) -> Result<Container> {
+    let all_containers = list_containers(channel).await?;
+    let cont = all_containers
+        .into_iter()
+        .find(|c| c.name == name)
+        .ok_or("Container not found")?;
+
+    Ok(cont)
+}
+
+pub async fn start_container(channel: &mut ClientChannel, id: &str) -> Result<()> {
+    let _r = tonic::Request::new(cm_rpc::StartContainerRequest {
+        id: String::from(id),
+    });
+    let _r = channel.start(_r).await?;
+    Ok(())
+}
+
+pub async fn stop_container(channel: &mut ClientChannel, id: &str, timeout: i64) -> Result<()> {
+    let stop_options = Some(cm_types::StopOptions {
+        timeout,
+        force: true,
+        signal: String::from("SIGTERM"),
+    });
+
+    let _r = tonic::Request::new(cm_rpc::StopContainerRequest {
+        id: String::from(id),
+        stop_options,
+    });
+    let _r = channel.stop(_r).await?;
+    Ok(())
+}
+
+pub async fn remove_container(channel: &mut ClientChannel, id: &str, force: bool) -> Result<()> {
+    let _r = tonic::Request::new(cm_rpc::RemoveContainerRequest {
+        id: String::from(id),
+        force,
+    });
+    let _r = channel.remove(_r).await?;
+    Ok(())
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+struct KantoLogLine {
+    stream: String,
+    log: String,
+    time: String,
+}
+
+// Strips the console control characters and updates the log string
+fn strip_and_push(line: &KantoLogLine, log: &mut String) {
+    let stripped: Vec<u8> = strip(line.log.clone()).unwrap();
+    log.push_str(String::from_utf8_lossy(&stripped).as_ref());
+}
+
+// Warning! This function currently uses system paths since the author is not aware of a way to obtains logs via grpc from CM.
```

Comment:
A feature request has been filed with the Eclipse Kanto project:
https://github.com/eclipse-kanto/container-management/issues/98